### PR TITLE
Scale mmt

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR optimizes MMT startup time.

Fixes #\<issue_number>

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--19984.org.readthedocs.build/en/19984/

cc @borda @carmocca @justusschock @awaelchli

### CODE REVIEW

File: src/lightning/app/core/queues.py [hunk 1]
```diff
@@ -520,20 +522,25 @@ def batch_get(self, timeout: Optional[float] = None, count: Optional[int] = None
...
 
-        value = pickle.dumps(item)
+        value = pickle.dumps(item, protocol=pickle.HIGHEST_PROTOCOL)
         queue_len = self.length()
         if queue_len >= WARNING_QUEUE_SIZE:
             warnings.warn(
                 f"The Queue {self._name_suffix} length is larger than the recommended length of {WARNING_QUEUE_SIZE}. "
                 f"Found {queue_len}. This might cause your application to crash, please investigate this."
             )
+        print("AAA", time.time() - t0)
```
Comment (lantiga) [inline src/lightning/app/core/queues.py:536]:
```suggestion
```

File: src/lightning/app/core/queues.py [hunk 2]
```diff
@@ -520,20 +522,25 @@ def batch_get(self, timeout: Optional[float] = None, count: Optional[int] = None
...
         resp = self.client.post(f"v1/{self.app_id}/{self._name_suffix}", data=value, query_params={"action": "push"})
+        print("BBB", time.time() - t0)
```
Comment (lantiga) [inline src/lightning/app/core/queues.py:538]:
```suggestion
```

File: src/lightning/app/core/queues.py [hunk 3]
```diff
@@ -520,20 +522,25 @@ def batch_get(self, timeout: Optional[float] = None, count: Optional[int] = None
...
         if resp.status_code != 201:
             raise RuntimeError(f"Failed to push to queue: {self._name_suffix}")
 
+        print("CCC", time.time() - t0)
```
Comment (lantiga) [inline src/lightning/app/core/queues.py:542]:
```suggestion
```

File: src/lightning/app/core/queues.py [hunk 4]
```diff
@@ -520,20 +522,25 @@ def batch_get(self, timeout: Optional[float] = None, count: Optional[int] = None
 
     @backoff.on_exception(backoff.expo, (RuntimeError, requests.exceptions.HTTPError))
     def put(self, item: Any) -> None:
+        t0 = time.time()
```
Comment (lantiga) [inline src/lightning/app/core/queues.py:525]:
```suggestion
```

**Review by lantiga — APPROVED** [review]

File: src/lightning/app/core/queues.py (hunk 1 above)
Comment (tchaton) [reply src/lightning/app/core/queues.py:536]: Still debugging ;)

**Review by justusschock — APPROVED** [review]

**Pull request merged**